command/job_stop.go

```diff
@@ -140,92 +140,141 @@ func (c *JobStopCommand) Run(args []string) int {
 		return 1
 	}
 
-	// Check if the job exists
-	jobs, _, err := client.Jobs().PrefixList(jobID)
-	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Error deregistering job: %s", err))
-		return 1
-	}
-	if len(jobs) == 0 {
-		c.Ui.Error(fmt.Sprintf("No job(s) with prefix or id %q found", jobID))
-		return 1
+	// TODO: since we're printing errors as
+	// we receive them, we don't need a new ch
+	// errorCh := make(chan error, len(jobIDs))
+
+	statusCh := make(chan int, len(jobIDs))
+
+	var wg sync.WaitGroup
+	for _, jobID := range jobIDs {
+		jobID := jobID
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			// Truncate the id unless full length is requested
+			length := shortId
+			if verbose {
+				length = fullId
+			}
+
+			// Check if the job exists
+			jobs, _, err := client.Jobs().PrefixList(jobID)
+			if err != nil {
+				// TODO: can we improve this error message?
+				c.Ui.Error(fmt.Sprintf("Error deregistering job with id %s err: %s", jobID, err))
+				statusCh <- 1
+				return
+			}
+			if len(jobs) == 0 {
+				c.Ui.Error(fmt.Sprintf("No job(s) with prefix or id %q found", jobID))
+				statusCh <- 1
+				return
+			}
+			if len(jobs) > 1 {
+				if (jobID != jobs[0].ID) || (c.allNamespaces() && jobs[0].ID == jobs[1].ID) {
+					c.Ui.Error(fmt.Sprintf("Prefix matched multiple jobs\n\n%s", createStatusListOutput(jobs, c.allNamespaces())))
+					statusCh <- 1
+					return
+				}
+			}
+
+			// Prefix lookup matched a single job
+			q := &api.QueryOptions{Namespace: jobs[0].JobSummary.Namespace}
+			job, _, err := client.Jobs().Info(jobs[0].ID, q)
+			if err != nil {
+				c.Ui.Error(fmt.Sprintf("Error deregistering job with id %s err: %s", jobID, err))
+				statusCh <- 1
+				return
+			}
+
+			// TODO: does confirmation make sense with >1 jobs?
```

Comment:
Need suggestions here, confirmation certainly doesn't seem convenient for >1 jobs but is a healthy behavior for when we have only one job. What do you think?

Reply:
Agreed. Other goroutines may be logging output at the same time which would create quite the confusing mess for users. Best to treat batch stops as a non-interactive command in the sense that we shouldn't expect user input after they've run it.